src/coord/src/coord.rs

```diff
@@ -1129,7 +1143,10 @@ impl<S: Append + 'static> Coordinator<S> {
                     .expect("sending to internal_cmd_tx cannot fail");
             });
         } else {
-            self.group_commit().await;
+            match Arc::clone(&self.write_lock).try_lock_owned() {
+                Ok(_guard) => self.group_commit().await,
```

Comment:
I do not like that `group_commit` is now called in two different places. But I don't have a better suggestion.

Reply:
You might prefer this approach which only calls `group_commit` from a single place: https://github.com/MaterializeInc/materialize/pull/13206/commits/0f6684a2b8eb3570894c5299d5a9859afd7bf4ed (It also passes CI).

The tradeoff is that some of the `defer_write` logic is duplicated.